### ISSUE TITLE
Provide a context key to check if LSP has a specific capability

### ISSUE THREAD

**Issue by randy3k** [sublimelsp/LSP#765]
<del>
Possible use case of the context key is to define custom keybind for go to definitions.
With the context key, it allows me to bind `super+alt+down` to both `goto_definition` and `lsp_symbol_definition`.

```js
{ 
   // this is the default key for `goto_definition`
  "keys": ["super+alt+down"], "command": "goto_definition" 
},
{ 
  "keys": ["super+alt+down"], "command": "lsp_symbol_definition" ,
  "context": [
			{ "key": "lsp.is.active", "operator": "equal", "operand": true }
		]
}
```
</del>

Update: @rwols pointed out that there is `lsp_active` to check, so this issue is now requesting a key for checking more specific capability.

**Comment by randy3k** [comment]
I got my hand dirty and made this

```
import sublime_plugin
from LSP.plugin.core.registry import session_for_view


class LspCapability(sublime_plugin.EventListener):
    def on_query_context(self, view, key, operator, operand, match_all):
        if key == "lsp.client_with_capability":
            return session_for_view(view, operand)
```

and now I could do
```
    {
      "keys": ["super+alt+down"], "command": "lsp_symbol_definition" ,
      "context": [
                { "key": "lsp.client_with_capability", "operator": "equal", "operand": "definitionProvider" }
            ]
    }
```

Any interests in getting this merged?

**Comment by predragnikolic** [comment]
Pretty cool :-)

**Comment by jfcherng** [comment]
https://lsp.readthedocs.io/en/latest/features/

![image](https://user-images.githubusercontent.com/6594915/68409208-a6625700-01c1-11ea-973d-9ea0a2329c58.png)

Isn't this already the default behavior?

**Comment by randy3k** [comment]
In a non lsp document, `lsp_symbol_definition` is [disabled](https://github.com/tomv564/LSP/blob/master/plugin/goto.py#L26), hence `f12` is simply malfunction.  It only fallbacks to `goto_definition` when lsp is at work.

**Comment by jfcherng** [comment]
You are right... https://github.com/tomv564/LSP/blob/be60102/plugin/goto.py#L28-L58
No wonder sometimes I get my F12 not working :open_mouth:

**Comment by rwols** [comment]
Related: https://github.com/tomv564/LSP/issues/651, https://github.com/tomv564/LSP/commit/88fc8f9786569aab2a09a62f873af7e46aaef025

I think this already works with `lsp_active`, no?

**Comment by jfcherng** [comment]
@rwols I can confirm that `lsp_active` works. Thanks.

```js
{ "keys": ["f12"], "command": "lsp_symbol_definition", "context": [{"key": "setting.lsp_active"}]},
```

Maybe this is worth written in the docs :)

**Comment by randy3k** [comment]
Though `lsp_active` is good enough for my purpose, the context key I posted above also checks if the client has such capability.


> ```js
> { "keys": ["f12"], "command": "lsp_symbol_definition", "context": [{"key": "setting.lsp_active"}]},
> ```
> 
> Maybe this is worth written in the docs :)

Perhaps, we should put the context in all [these keyblindings](https://github.com/tomv564/LSP/blob/master/Keymaps/Default%20(OSX).sublime-keymap#L28-L56)

**Comment by randy3k** [comment]
I changed the original post and the title for a key checking a more specific capability. (Though I don't need anymore with `lsp_active`) :)

**Comment by rwols** [comment]
Fixed by #1142 for ST4. Check out the new keybinding contexts.
